app/src/auth.py

```diff
@@ -32,7 +32,7 @@ def login_post():
         account = teacher
     elif user and not teacher:
         account = user
-    else:
+    elif user and teacher:
         flash('The email address is associated with multiple roles. Please contact support.', 'login_error')
         return redirect(url_for('auth.login'))
```

Comment:
Before this if statement (line 29-ish), add a line instantiating account such that `account = None`. Otherwise, if none of the conditionals here are triggered, the server will crash at the line 41 conditional due to the variable not existing before being used.

Reply:
Sounds good, I've made the change.